### DISCUSSION TITLE
Calculate a seperate fee in checkout

### DISCUSSION THREAD

**Discussion by sstottelaar** [duncanmcclean/simple-commerce#1172] [Help]
Hi,

I’m building a shop in that sells heavy electronic equipment. In the Netherlands, we have a required “disposal fee” (an additional tax) for electronics, which is calculated based on factors like product weight. Each product has a unique disposal fee, and my client would like this fee to appear separately at checkout rather than being included in the product price.

I’ve considered creating a computed field to add the disposal fee to the product price, but this would mean the fee isn’t reflected in the cart’s total calculations. I also thought about extending the tax engine to include the disposal fee, but this could interfere with the tax calculations. An extra line item would also suffice, but that requires a lot of custom work (I think) in the process of adding/removing items from the cart.

Does anyone have suggestions on the best way to handle this? Any help would be much appreciated!

**Comment by duncanmcclean** [comment]
The easiest way would probably be to add the "disposal fee" as a line item, then you don't have to mess around with getting the grand total to reflect your extra total.

You could maybe listen to the [`OrderSaved`](https://simple-commerce.duncanmcclean.com/events#content-ordersaved) event, check that the disposal fee line item hasn't already been added, and [add the line item programatically](https://simple-commerce.duncanmcclean.com/php-api#content-orders).
